trunk/SUAVE/Components/Energy/Networks/Turboelectric_HTS_Ducted_Fan.py

```diff
@@ -0,0 +1,193 @@
+        ambient_skin                = self.ambient_skin             # flag to indicate rotor skin temp
+        rotor_surface_temp          = self.skin_temp                # Exterior temperature of the rotors
+        leads                       = self.leads                    # number of rotor leads, typically twice the number of rotors
+        number_of_engines           = self.number_of_engines        # number of propulsors and number of propulsion motors
+        number_of_supplies          = self.powersupply.number_of_engines    # number of turboelectric generators
+
+       
+    
+        conditions      = state.conditions
+        numerics        = state.numerics
+
+        amb_temp        = conditions.freestream.temperature
+
+        # Solve the thrust using the other network (i.e. the ducted fan network)
+        results = ducted_fan.evaluate_thrust(state)
+
+        # Calculate the required electric power to be supplied to the ducted fan motor by dividing the shaft power required by the ducted fan by the efficiency of the ducted fan motor
+        # Note here that the efficiency must not include the efficiency of the rotor and rotor supply components as these are handled separately below.
+        # powersupply.inputs.power_in = propulsor.thrust.outputs.power/motor.motor_efficiency
+        motor_power_in        = ducted_fan.thrust.outputs.power/motor.motor_efficiency
+
+        # Calculate the power used by the power electronics. This does not include the power delivered by the power elctronics to the fan motor.
+        esc_power             = motor_power_in/esc.efficiency - motor_power_in
+
+        # Set the rotor skin temp. Either it's ambient, or it's the temperature set in the rotor.
+        skin_temp           = np.empty_like(amb_temp)
+        skin_temp[:]        = amb_temp
+        if ambient_skin == 0:
+            skin_temp[:]    = rotor_surface_temp 
+
+        # If the rotor current is to be varied depending on the motor power here is the place to do it. For now the rotor current is set as constant.
+        rotor_current       = np.full_like(motor_power_in, rotor.current)
+
+        # Calculate the power that must be supplied to the rotor. This also calculates the cryo load per rotor and stores this value as rotor.outputs.cryo_load
+        single_rotor_power  = rotor.power(rotor_current, skin_temp)
+        rotor_power_in      = single_rotor_power * ducted_fan.number_of_engines
+
+
+        # -------- Rotor Current Supply ---------------------------------
+
+
+        # Calculate the power loss in the rotor current supply leads.
+        # The cryogenic loading due to the leads is also calculated here.
+        lead_power            = np.zeros_like(rotor_current)
+        lead_cryo_load        = np.full_like(rotor_current, lead.unpowered_Q)
+        # Iterate through each element in the rotor_current array to build the cryo_load and lead_power array
+        for index, r_current in np.ndenumerate(rotor_current):
```

Comment:
this should be vectorizable. In SUAVE we try to never loop over time only over components.

Reply:
rotor_current is a list of currents to be supplied to each rotor.  I think this means I am not looping over time?

Reply:
Each current is sequential in time, so it is a loop over time steps. SUAVE is vectorized over time steps. The speed difference is enormous. The code for Q_offdesign is vectorizable. What that means is that the conditional statements can be placed into the arrays. We have example like this all over the code. But here's an example in another network: https://github.com/suavecode/SUAVE/blob/develop/trunk/SUAVE/Components/Energy/Networks/Battery_Propeller.py#L211